### PR TITLE
Enable syntax language server

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

It requires https://github.com/eclipse/eclipse.jdt.ls/pull/1386

This PR exposes a new user setting `java.server.launchMode` to control whether to enable syntax language server, below is the accepted values:
- Standard
Provides full features such as intellisense, refactoring, building, Maven/Gradle support etc.
- LightWeight
Starts a syntax server with lower start-up cost. Only provides syntax features such as outline, navigation, javadoc, syntax errors. The lightweight mode won't load thirdparty extensions, such as java test runner, java debugger, etc.
 - Hybrid
Provides full features with better responsiveness. It starts a standard language server and a secondary syntax server. The syntax server provides syntax features until the standard server is ready. And the syntax server will be shutdown automatically after the standard server is fully ready.

### CODE REVIEW

**Comment by testforstephen** [comment]
The test failure is because it downloads the snapshot from the remote site, and it's not compatible with new changes. After the eclipse/eclipse.jdt.ls#1386 PR is merged, this failure will be auto resolved.

@fbricon @snjeza @akaroml Could you take a look at the PR?

File: package.json [hunk 1]
```diff
@@ -493,6 +493,22 @@
           "default": [],
           "description": "Java Execution Environments -> Runtimes.",
           "scope": "machine"
+        },
+        "java.server.launchMode": {
+          "type": "string",
+          "enum": [
+            "Standard",
+            "LightWeight",
+            "HighPerformance"
```
Comment (akaroml) [inline package.json]: HighPerformance looks like Hybrid to me.

File: src/settings.ts [hunk 2]
```diff
@@ -191,4 +192,15 @@ export function getJavaagentFlag(vmargs) {
...
 	}
 	return agentFlag;
-}
\ No newline at end of file
+}
+
+export enum ServerMode {
+	STANDARD = 'Standard',
+	LIGHTWEIGHT = 'LightWeight',
+	HIGHPERFORMANCE = 'HighPerformance'
```
Comment (akaroml) [inline src/settings.ts]: Hybrid?

File: package.json (hunk 1 above)
Comment (testforstephen) [reply package.json]: yeah, it's Hybrid.

**Pull request merged**